src/client/pages/Checkout/CheckoutPayment/ContactInformation/ContactInformation.tsx

```diff
@@ -32,89 +41,94 @@ const InputFieldsWrapper = styled.div`
   }
 `
 
-const StyledInputField = styled(InputField)`
-  ${Wrapper} & {
-    background-color: ${colorsV3.gray100};
-    border-radius: 8px;
-  }
-`
-
 const SpacerSmall = styled.div`
   height: 1rem;
 `
 
-const Spacer = styled.div`
-  height: 1.5rem;
-`
-
 const HorizontalDivider = styled(Divider)`
   margin: 1.5rem 0;
   ${MEDIUM_SMALL_SCREEN_MEDIA_QUERY} {
     margin: 3rem 0;
   }
 `
 
-export const ContactInformation = ({
-  firstName,
-  lastName,
-  email,
-  ssn,
-}: ContactInfoData) => {
-  const textKeys = useTextKeys()
-  const locale = useCurrentLocale()
+type Props = {
+  formikProps: FormikProps<QuoteInput>
+}
 
+export const ContactInformation = ({ formikProps }: Props) => {
+  const textKeys = useTextKeys()
+  const { handleChange } = formikProps
   const [hasEnabledCreditCheckInfo] = useFeature([
     Features.CHECKOUT_CREDIT_CHECK,
   ])
-  const ssnFormatExample = locale.ssn.formatExample
+  const [isShowingCreditCheckInfo, setIsShowingCreditCheckInfo] = useState(
+    false,
+  )
+  const {
+    values: { ssn },
+    initialValues,
+    submitForm,
+  } = formikProps
+
+  const debouncedSubmitForm = useCallback(
```

Comment:
How come this is needed? Doesn't `disabled={isLoading}` on the Footer Button make it impossible to click many times?

Reply:
There is a useEffect below that submits the form on SSN change. We submit the SSN immediately (without waiting for users to click the Footer Button, because SSN changes can affect the price. So they need to see the new price before they click continue.

Reply:
The debouncing is to make sure when users type in the SSN, that the form is not submitted many times